### PR TITLE
BIP155 follow-ups

### PR DESCRIPTION
**Pull request by sipa**
This:
* Documents BIP155 support in doc/bips.md
* Restricts addrv2 relay to IPv4, IPv6, and Tor addresses. Relaying addresses in ranges that no network software has support for seems like a gratuitous spam vector.

### CODE REVIEW

**Review by hebasto — COMMENTED** [review]
Concept ACK.

File: doc/bips.md [hunk 1]
```diff
@@ -37,7 +37,8 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v0.21.0**):
...
 * [`BIP 147`](https://github.com/bitcoin/bips/blob/master/bip-0147.mediawiki): NULLDUMMY softfork as of **v0.13.1** ([PR 8636](https://github.com/bitcoin/bitcoin/pull/8636) and [PR 8937](https://github.com/bitcoin/bitcoin/pull/8937)), *buried* since **v0.19.0** ([PR #16060](https://github.com/bitcoin/bitcoin/pull/16060)).
 * [`BIP 152`](https://github.com/bitcoin/bips/blob/master/bip-0152.mediawiki): Compact block transfer and related optimizations are used as of **v0.13.0** ([PR 8068](https://github.com/bitcoin/bitcoin/pull/8068)).
-- [`BIP 158`](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki): Compact Block Filters for Light Clients can be indexed as of **v0.19.0** ([PR #14121](https://github.com/bitcoin/bitcoin/pull/14121)).
+* [`BIP 155`](https://github.com/bitcoin/bips/blob/master/bip-0155.mediawiki): The 'addrv2' and 'sendaddrv2' messages which enable relay of Tor V3 addresses and more are supported as of **v0.21.0** ([PR 19845](https://github.com/bitcoin/bitcoin/pull/19845)).
```
Comment (hebasto) [inline doc/bips.md:40]: "and more are supported" ?

**Review by hebasto — APPROVED** [review]
ACK 7ca0e87278ffc096df28c6779bb2a8d2d10161f9

File: doc/bips.md [hunk 2]
```diff
@@ -37,7 +37,8 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v0.21.0**):
...
+* [`BIP 158`](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki): Compact Block Filters for Light Clients can be indexed as of **v0.19.0** ([PR #14121](https://github.com/bitcoin/bitcoin/pull/14121)).
```
Comment (jonatack) [inline doc/bips.md:41]: ISTM should be [PR 19954](https://github.com/bitcoin/bitcoin/pull/19954) (or maybe 19031)

File: doc/bips.md (hunk 1 above)
Comment (jonatack) [inline doc/bips.md:40]: ISTM should be `[PR 19954](https://github.com/bitcoin/bitcoin/pull/19954)` (or maybe 19031)

**Comment by jonatack** [comment]
Concept ACK.

It's unclear to me why the additional "should we relay this address?" logic is in `RelayAddress()` instead in the caller along with the existing logic. Would it better for the logic to be in one place?

net_processing.cpp::L2660
```cpp
            bool fReachable = IsReachable(addr);
            if (addr.nTime > nSince && !pfrom.fGetAddr && vAddr.size() <= 10 && addr.IsRoutable())
            {
                // Relay to a limited number of other nodes
                RelayAddress(addr, fReachable, m_connman);
            }
```

File: doc/bips.md (hunk 1 above)
Comment (sipa) [reply doc/bips.md]: Fixed, it should have been #19954.

**Comment by sipa** [comment]
@jonatack I prefer to avoid having explicit network-dependent code outside of net/netaddress.

**Comment by jonatack** [comment]
ACK 56f9dba015c592b8925795012e3061a710070a27

File: src/net_processing.cpp [hunk 3]
```diff
@@ -1528,6 +1528,7 @@ void RelayTransaction(const uint256& txid, const uint256& wtxid, const CConnman&
 
 static void RelayAddress(const CAddress& addr, bool fReachable, const CConnman& connman)
 {
+    if (!fReachable && !addr.IsRelayable()) return;
```
Comment (hebasto) [inline src/net_processing.cpp:1531]: The following code https://github.com/bitcoin/bitcoin/blob/56f9dba015c592b8925795012e3061a710070a27/src/net_processing.cpp#L1540-L1541

does not work for `!fReachable` addresses now.

At least, the comment is wrong now, right?

**Review by hebasto — APPROVED** [review]
ACK 56f9dba015c592b8925795012e3061a710070a27, verified both links.

**Comment by naumenkogs** [comment]
ACK 56f9dba

**Review by vasild — COMMENTED** [review]
(post merge) ACK 56f9dba01

This patch achieves the same as the one at the bottom of https://github.com/bitcoin/bitcoin/pull/19954#issuecomment-701307791 and in addition the patch in this PR will relay I2P addresses if they become reachable in the future without needing an adjustment.

Given that the newly added function `IsRelayable()` deals with a higher level logic of whether to relay or not, maybe it would be better if it lives somewhere in `net_processing.cc`, next to `RelayAddress()` instead of as a method of the lower level `CNetAddr` class. But it works anyway and can be adjusted further if needed.

Thanks!

**Comment by jonatack** [comment]
Further BIP155 follow-ups and minor bugfix at #20120.

**Pull request merged**